pkg/registry/networking/networkpolicy/storage/storage.go

```diff
@@ -72,36 +62,3 @@ var _ rest.ShortNamesProvider = &REST{}
 func (r *REST) ShortNames() []string {
 	return []string{"netpol"}
 }
-
-// StatusREST implements the REST endpoint for changing the status of an ingress
-type StatusREST struct {
-	store *genericregistry.Store
-}
-
-// New creates an instance of the StatusREST object
-func (r *StatusREST) New() runtime.Object {
-	return &networkingapi.NetworkPolicy{}
-}
-
-// Destroy cleans up resources on shutdown.
-func (r *StatusREST) Destroy() {
-	// Given that underlying store is shared with REST,
-	// we don't destroy it here explicitly.
-}
```

Comment:
ok... so I assume this means that removing the `StatusREST` object will not potentially cause old status data to be orphaned in etcd?

Reply:
It will, and this is why we are stating that a different protobuf (name or id, not sure) should be used.

Reply:
no, I meant, if you create a NetworkPolicy object with 1.26+featuregate and it gets a `status` field, and then you upgrade to 1.27 and delete the NetworkPolicy object, does that leak any data in etcd? I'm assuming "no", but I'm not _convinced_ the answer is "no"

Reply:
if the question is about this Destroy() method, it was added to shutdown cleanly the apiserver https://github.com/kubernetes/kubernetes/pull/109303. , is not that it is going to wipe out data in etcd

Reply:
> no, I meant, if you create a NetworkPolicy object with 1.26+featuregate and it gets a `status` field, and then you upgrade to 1.27 and delete the NetworkPolicy object, does that leak any data in etcd? I'm assuming "no", but I'm not _convinced_ the answer is "no"

it is the star thing, no? it will convert it to the current version and replace it on write https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts.html

Reply:
I guess I'm not explaining this well.

So like, if we had added a new top-level BrokenNetworkPolicy object in 1.26, then decided it was a bad idea and removed it in 1.27, we would need to do extra work on 1.26→1.27 upgrade to delete any BrokenNetworkPolicy objects in etcd, or they'd get leaked forever. (Right?)

Now, we are not removing an entire type here, but we aren't just removing a field from a type either; we are removing a big chunk of code from `pkg/registry/.../storage/` that presumably has _something to do with how stuff is stored in the registry_. So my question is, will removing this code cause the apiserver to lose track of any data that it had previously stored in etcd?

Reply:
> So like, if we had added a new top-level BrokenNetworkPolicy object in 1.26, then decided it was a bad idea and removed it in 1.27, we would need to do extra work on 1.26→1.27 upgrade to delete any BrokenNetworkPolicy objects in etcd, or they'd get leaked forever. (Right?)

That's correct